src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/CompareSymbolsCorrectlyFix.cs

```diff
@@ -38,6 +41,15 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
                             equivalenceKey: nameof(CompareSymbolsCorrectlyFix)),
                         diagnostic);
                 }
+                else if (diagnostic.Descriptor == CompareSymbolsCorrectlyAnalyzer.CollectionRule)
+                {
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            CodeAnalysisDiagnosticsResources.CompareSymbolsCorrectlyCodeFix,
+                            cancellationToken => CallOverloadWithEqualityComparerAsync(context.Document, diagnostic.Location.SourceSpan, cancellationToken),
+                            equivalenceKey: nameof(CompareSymbolsCorrectlyFix)),
```

Comment:
Are we purposefully using the same equivalence key as for prior fix that converts to equals? Note that this will cause FixAll for a diagnostic triggered for one of these rules to also fix all instances of other rule. I think you want to probably use a different equivalence key?

Reply:
I think I didn't think it through enough but somehow I don't see why people willing to fix `Equals` would not want to fix the collection at the same time. If there was a way to differentiate the rule I would think splitting the code-fix would make sense but it's not the case.

I'll let you have the final word, so please let me know your preference.

Reply:
Fine with keeping this as-is. If we get complaints, we can split.

Reply:
Fine with keeping this as-is. If we get complaints, we can split.